### PR TITLE
Embeds xhyve

### PR DESCRIPTION
**Pull request by johanneswuerbach**
Embeds xhyve golang bindings

Uses https://github.com/hooklift/xhyve, which is also used by https://github.com/TheNewNormal/corectl

### CODE REVIEW

**Comment by zchee** [comment]
@johanneswuerbach Thanks PR!

Yes, I knew that repository. 
but, Now c4milo seems to have developed here.
https://github.com/xhyve-xyz/xhyve

I think we should include that repository.

How do you think?

**Comment by johanneswuerbach** [comment]
As far as I understand https://github.com/xhyve-xyz/xhyve has no go bindings, just xhyve and some patches.

Looks like https://github.com/hooklift/xhyve/pull/5 there is already work in progress to update the gobindings from the repo you mentioned.

**Comment by zchee** [comment]
@johanneswuerbach 
Yes.
But, If the development of the https://github.com/xhyve-xyz/xhyve has been accelerated, not promise that `libxhyve` is follow up.
Despite the new features have been added to the https://github.com/xhyve-xyz/xhyve, but we might not use it.
I think to fork the code of libxhyve inside docker-machine-driver-xhyve(not all), also to submodule control the xhyve-xyz.
How do you think?

BTW, this PR does not run xhyve VM.
I want to put a comment, but I can not browse due there is an updated vendor of git log.

Could you rebase https://github.com/zchee/docker-machine-driver-xhyve/pull/28 and separating commit of `xhyve.go` and vendor?

**Comment by johanneswuerbach** [comment]
You are right, this breaks the driver. I'll revisit when I have time, rebased for now.

> I think to fork the code of libxhyve inside docker-machine-driver-xhyve(not all), also to submodule control the xhyve-xyz.

Dunno, I'm usually not really a fan of forking and duplicating, but your call. Maybe we could ask the creators of `xhyve-xyz` to provide an "offical" binding? @AntonioMeireles @c4milo @jeremyhu

**Comment by jeremyhu** [comment]
FWIW, the goal of xhyve-xyz is to have a central place to work on xhyve rather than seeing the proliferation of different forks for different needs.  I'd much prefer to see work like this inside the main xhyve repo.

I like the idea of making xhyve into a library that can have bindings for different languages and a thin CLI on top of it.

**Comment by c4milo** [comment]
There are a couple of reasons https://github.com/hooklift/xhyve exists:

*  We needed to flatten out the project in order to statically compile the library using Go tooling. Before this we tried creating a static library of xhyve, linking it from Go using CGO but xhyve did't work correctly when doing this. 
* We make some calls from C into Go to clean up resources in Go land, as well as to let Go know about serial consoles created so you can connect to the VM from Go.
* We are also aiming to keep the Go library very lean, eventually there will be features people wants to get into xhyve upstream that would not make sense to have them in the Go library and vice-versa.

**Comment by johanneswuerbach** [comment]
@zchee updated and now it works for me :bow:

**Comment by zchee** [comment]
@johanneswuerbach 
mist64, He came back.
I will wait and see it.
Sorry :(

File: mk/driver.mk [hunk 1]
```diff
@@ -55,7 +55,7 @@ test-url:
 	${DOCKER_MACHINE_CMD} --storage-path ${DOCKER_MACHINE_STORAGEPATH} url ${DOCKER_MACHINE_VM_NAME}
 
 driver-run: clean build install driver-kill
-	rm -rf ${DOCKER_MACHINE_STORAGEPATH}/machines/${DOCKER_MACHINE_VM_NAME} && ${DOCKER_MACHINE_CMD} --storage-path ${DOCKER_MACHINE_STORAGEPATH} create --driver xhyve --xhyve-disk-size ${DOCKER_MACHINE_VM_DISKSIZE} ${DOCKER_MACHINE_VM_NAME}
+	${DOCKER_MACHINE_CMD} --storage-path ${DOCKER_MACHINE_STORAGEPATH} rm -f ${DOCKER_MACHINE_VM_NAME} && ${DOCKER_MACHINE_CMD} --storage-path ${DOCKER_MACHINE_STORAGEPATH} create --driver xhyve --xhyve-disk-size ${DOCKER_MACHINE_VM_DISKSIZE} ${DOCKER_MACHINE_VM_NAME}
```
Comment (zchee) [inline mk/driver.mk]: @johanneswuerbach
If does not exists `xhyve-test` VM, occurred error.

Solution
```
${DOCKER_MACHINE_CMD} --storage-path ${DOCKER_MACHINE_STORAGEPATH} rm -f ${DOCKER_MACHINE_VM_NAME} || true
${DOCKER_MACHINE_CMD} --storage-path ${DOCKER_MACHINE_STORAGEPATH} create --driver xhyve --xhyve-disk-size ${DOCKER_MACHINE_VM_DISKSIZE} ${DOCKER_MACHINE_VM_NAME}
```

**Comment by zchee** [comment]
I have the discussion with @johanneswuerbach in the twitter.
For now merges.

LGTM.

**Pull request merged**